apps/vc-api/src/key/key.controller.ts

```diff
@@ -35,21 +45,29 @@ export class KeyController {
   @Post()
   @ApiBody({ type: KeyPairDto })
   @ApiCreatedResponse({ type: KeyDescriptionDto })
+  @ApiBadRequestResponse({ type: BadRequestResponseDto })
```

Comment:
@artursudnik how would you compare the approach you've implemented here vs https://nanogiants.github.io/nestjs-swagger-api-exception-decorator/ ?

Reply:
This is for sure going to be useful after custom exceptions are thrown. I am not 100% sure if it will help at this stage, where mainly `BadRequestException` is thrown by the `class-validator` and NestJS. I plan to look into this later to see if it allows for simplifying things.

Reply:
At least it would allow getting rid of error DTOs I've defined. I am not sure yet...

Reply:
> At least it would allow getting rid of error DTOs I've defined.

Yes, I think the main advantage is that the DTOs could be removed.

>  I am not 100% sure if it will help at this stage, where mainly BadRequestException is thrown by the class-validator and NestJS

I'm not really getting this point. `BadRequestException` is one of the example in the package documentation: https://nanogiants.github.io/nestjs-swagger-api-exception-decorator/gettingstarted/usage/simple#route-methods so I think this package could help with this exception as well. Could you elaborate a bit please?

Reply:
What I mean is that BadRequestException is thrown by the NestJS validation mechanism, so in this case, it is a very standardized situation and we will not benefit too much from using any wrapper that helps with casting different exception types to one type of HTTP error response. If we for some reason decide later to extend BadRequestException and have multiple types it would make sense, but now I'd start with cleaning things to see if we need this.

Reply:
Ok, sounds good

Reply:
I played around with https://nanogiants.github.io/nestjs-swagger-api-exception-decorator/  a bit more. It seems to work well and could maybe reduce the boilerplate Error DTO code.

The main limitation that I see at the moment is that it doesn't seem possible to specify a custom schema. So I don't think we can do `message` being `oneOf` `string` or `array`.

I guess it makes sense to just keep it as then @artursudnik (with the custom DTOs and without the use of https://nanogiants.github.io/nestjs-swagger-api-exception-decorator/ )?

Reply:
Yes, I think that using this would not solve any problem now.